spec/generators/sorbet_test_cases.rb

```diff
@@ -305,6 +305,17 @@
 T.assert_type!(Wizard::House::Ravenclaw, Wizard::House)
 T.assert_type!(Wizard::House::Slytherin, Wizard::House)
 
+# Serialization
+T.assert_type!(
+  wizard.owl_results,
+  T.any(T::Array[T.untyped], T::Boolean, Float, T::Hash[T.untyped, T.untyped], Integer, String)
```

Comment:
It looks like `owl_results` is serialized as a Hash. I'd expect the type here to be `T::Hash[T.untyped, T.untyped]`. Why do we also have other possible type like Integer, String, etc. here?
```
  serialize :owl_results, Hash
```

Reply:
In the generated RBI, it seems the type is only `T.nilable(T::Hash[...])`
```
  sig { returns(T.nilable(T::Hash[T.untyped, T.untyped])) }
  def owl_results; end
```

Reply:
Yes, indeed.  Thanks. I've fixed up the `nil` handling throughout and corrected that incorrect `Hash` type.

Locally, I'd created a version where some of the serialised columns where nullable and some not.  I ran the tests and got a bunch of errors with `Wizard.create!` etc... since they now expected some non-null columns. 

Rather than update all the test factories to pass (and making this PR bigger) I instead just let all serialised columns by nullable.  If you'd prefer that another way lmk and I'm happy to make some changes.